src/components/AllComponents.js

```diff
@@ -78,52 +70,8 @@ function AllComponents() {
     });
   },[onLoadGetUserCharacters])
 
-  useEffect(() => {
-    setCharacters(allCharacters);
-    setFilteredCharacters(allCharacters);
-  }, [allCharacters]);
-
-  // useEffect(() => {
-  //   if (!(characterCategory === "" || characterCategory === "All Categories")) {
-  //     setFilteredCharacters(
-  //       filteredCharacters.filter((character) =>
-  //         character.category.includes(characterCategory)
-  //       )
-  //     );
-  //   }
-  //   if (rarityCategory !== "ALL") {
-  //     setFilteredCharacters(
-  //       filteredCharacters.filter(
-  //         (character) => character.rarity === rarityCategory
-  //       )
-  //     );
-  //   }
-  //   if (typeCharacters !== "ALL") {
-  //     setFilteredCharacters(
-  //       filteredCharacters.filter((character) =>
-  //         character.type.includes(typeCharacters)
-  //       )
-  //     );
-  //   }
-  //   if (!(search === "" || search === null)) {
-  //     setFilteredCharacters(
-  //       filteredCharacters.filter((character) =>
-  //         character.name.includes(search)
-  //       )
-  //     );
-  //   }
-  // }, [search, characterCategory, rarityCategory, typeCharacters]);
-
-
-
-
-
-  let suggestionArr = [];
-
-  function arraySuggestion(character) {
-    suggestionArr.push(character.id);
-    suggestionArr.push(character.link_skill);
-    setSuggestion(suggestionArr);
+  const addSuggestion = (character) => {
```

Comment:
Did a little cleanup here - you should pretty much never create an array in a component function body then edit the array for state/event handlers. I think this change should get at the intent of what you were going for (add new char data to the "suggestion" state array